packages/playout-gateway/src/atemUploader.ts

```diff
@@ -39,29 +40,44 @@ export class AtemUploadScript {
 		})
 	}
 
-	public loadFile(): Promise<Buffer> {
+	public loadFile(fileUrl: string): Promise<Buffer> {
 		return new Promise<Buffer>((resolve, reject) => {
-			fs.readFile(this.fileUrl, (e, data) => {
+			fs.readFile(fileUrl, (e, data) => {
 				consoleLog('got file')
 				if (e) reject(e)
 				else resolve(data)
 			})
 		})
 	}
+	public async loadFiles(folder: string): Promise<Buffer[]> {
+		const files = await fs.promises.readdir(folder)
+		consoleLog(files)
+		const loadBuffers = files.map((file) => fs.promises.readFile(path.join(folder, file)))
```

Comment:
I think it is dangerous to load all of the frames at once, and is highly likely to run out of memory.

If we assume a 5s clip at 1080 (not unreasonable for a wipe with trailing audio) then we need 1gb of free ram to load in every raw frame at once.

To completely fill the 2me4k we would need ~5gb, and a constellation would take ~12gb.

That said, this might be good enough for now if we know our clips are going to be really short <500mb

Reply:
That's a good point, unfortunately not something that can be fixed in this PR as it is just how the API in atem-connection was designed. I don't plan to upload audio as part of any wipes currently, just masks and overlays, so practically we should never hit this issue. A good thing to pick up when we implement this in the media-manager though.

Reply:
good point, I didnt think about that

I've created https://github.com/nrkno/tv-automation-atem-connection/issues/110 as a reminder to do something about this